policy/lib/bundles.rego

```diff
@@ -15,83 +18,165 @@ disallowed_task_reference(tasks) := {task |
 # Returns a subset of tasks that use an empty bundle reference.
 empty_task_bundle_reference(tasks) := {task |
 	some task in tasks
-	bundle(task) == ""
+	refs.task_ref(task).bundle == ""
 }
 
 # Returns a subset of tasks that use bundle references not pinned to a digest.
 unpinned_task_bundle(tasks) := {task |
 	some task in tasks
-	ref := image.parse(bundle(task))
+	ref := image.parse(_bundle_ref(task, data["task-bundles"]))
 	ref.digest == ""
 }
 
+# Returns if the required task-bundles data is missing
+default missing_task_bundles_data := false
+
+missing_task_bundles_data {
+	count(data["task-bundles"]) == 0
+}
+
 # Returns a subset of tasks that use an acceptable bundle reference, but
 # an updated bundle reference exists.
 out_of_date_task_bundle(tasks) := {task |
 	some task in tasks
-	ref := image.parse(bundle(task))
-	collection := _collection(ref)
 
-	some match_index, out_of_date in collection
-	is_equal(out_of_date, ref)
-	match_index > 0
+	ref := image.parse(_bundle_ref(task, data["task-bundles"]))
+
+	_is_out_of_date(ref)
 }
 
 # Returns a subset of tasks that do not use an acceptable bundle reference.
 unacceptable_task_bundle(tasks) := {task |
 	some task in tasks
-	ref := image.parse(bundle(task))
-	collection := _collection(ref)
 
-	matches := [record |
-		some record in collection
-		is_equal(record, ref)
-	]
+	ref := image.parse(_bundle_ref(task, data["task-bundles"]))
 
-	count(matches) == 0
+	_is_unacceptable(ref)
 }
 
-# Returns if the required task-bundles data is missing
-default missing_task_bundles_data := false
+_is_unacceptable(ref) {
+	not _is_acceptable(ref)
+}
 
-missing_task_bundles_data {
-	count(data["task-bundles"]) == 0
+_is_unacceptable(ref) {
```

Comment:
It's a bit confusing the `_is_unacceptable` is not just the inverse of `_is_acceptable`. Maybe a misnomer? 

Looking at the definition of `_is_acceptable`, I think we should include the `_is_expired` there.

Reply:
I think I should rename `_is_acceptable` to `_record_exists`, it is confusing this way.